app/services/badge_granter.rb

```diff
@@ -445,7 +445,7 @@ def self.backfill(badge, opts = nil)
       next if row.staff && badge.awarded_for_trust_level?
 
       notification = send_notification(row.user_id, row.username, row.locale, badge)
-      badge.trigger_badge_granted_event(row.user_id)
+      UserBadge.trigger_user_badge_granted_event(badge.id, row.user_id)
```

Comment:
Don’t we have a way to find the `user_badge` from the `badge` and the `row.user_id`? It’s too bad to have to call a class method on `UserBadge` and provide it with a badge id and a user id whereas a `user_badge` object already knows those two objects :thinking:

Reply:
We have a way but it comes at the expense of having to query for the user_badge which we do not have to do here.